drivers/dma/dma_stm32.c

```diff
@@ -198,20 +201,26 @@ static int dma_stm32_get_periph_increment(enum dma_addr_adj increment,
 static int dma_stm32_configure(struct device *dev, u32_t id,
 			       struct dma_config *config)
 {
+
 	struct dma_stm32_data *data = dev->driver_data;
-	struct dma_stm32_stream *stream = &data->streams[id];
 	const struct dma_stm32_config *dev_config =
 					dev->config->config_info;
 	DMA_TypeDef *dma = (DMA_TypeDef *)dev_config->base;
+	struct dma_stm32_stream *stream = &data->streams[id - STREAM_OFFSET];
```

Comment:
Out of curiosity, why are we moving this line? General idea of initializations should be following the reverse christmas tree format, which means the long lines should go up rather than down.

Reply:
Ok, (the reason is that line has been moved several time,)
Restoring initial position